specifyweb/frontend/js_src/lib/queryresults.js

```diff
@@ -214,30 +217,32 @@ function auditedObjFormatter(fieldSpecs, model, localize) {
 
         addResult: function(result, table, resource) {
             var row = $('<tr class="query-result">').appendTo(table).data('resource', resource);
-            var href = resource.viewUrl();
+            var href = resource?.viewUrl();
             _.each(this.fieldSpecs, function(f, i) {
-                this.renderResult(row, f, href, result, i+1, this.format, resource);
+                this.renderResult(row, f, href, result, i, this.format, resource);
```

Comment:
This `i+1` business may depend on whether the record ids are being included in the query. I would check this very carefully.

Reply:
I moved the responsibility for handling i vs i+1 into renderResult

Reply:
I moved the responsibility for handling i vs i+1 into renderResult. Double checked it just now just in case